metadata-ingestion/src/datahub/ingestion/source/s3/__init__.py

```diff
@@ -558,6 +562,15 @@ def ingest_table(self, table_data: TableData) -> Iterable[MetadataWorkUnit]:
             platformSchema=OtherSchemaClass(rawSchema=""),
         )
         dataset_snapshot.aspects.append(schema_metadata)
+        if (
+            self.source_config.use_s3_bucket_tags
+            or self.source_config.use_s3_object_tags
+        ):
+            bucket = get_bucket_name(table_data.table_path)
+            key_prefix = get_key_prefix(table_data.table_path)
+            s3_tags = self.get_s3_tags(bucket, key_prefix, dataset_urn)
```

Comment:
If table_data.full_path != table_data.table_path, it represents that this is a folder level dataset.

Reply:
Added this check and made the key_prefix None when this case is true.